src/android/ContactAccessorSdk5.java

```diff
@@ -1854,6 +1859,39 @@ public boolean remove(String id) {
         return (result > 0) ? true : false;
     }
 
+    /**
+     * Gets birthday date from contact JSON object
+     * @param contact an object to get birthday from
+     * @return birthday or null, if the field isn't present or
+     *   is malformed in the contact
+     */
+    private Date getBirthday(JSONObject contact) {
+        try {
+            Long timestamp = contact.getLong("birthday");
+            return new Date(timestamp);
+        } catch (JSONException e) {
+            Log.d(LOG_TAG, "Could not get birthday: " + e.getMessage());
```

Comment:
As a general good practice, instead of appending the exception message to the error string, pass the exception object to the log method overload that takes a throwable as the third parameter. The exception message and stack trace will then be printed to logcat.

Also, it would be more appropriate to use Log.e() instead of Log.d() for this kind of unexpected error.